bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubAbstractDeviceData.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.internal;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * An abstract prototype for wrappers around JSON responses to JSON INFO or
+ * GET_LIVE_DATA requests
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public abstract class NeoHubAbstractDeviceData {
+
+    @SuppressWarnings("null")
```

Comment:
Why is this warning suppressed here?

Reply:
This is due to a conflict between the Eclipse compiler and the MVN command line compiler. The outer class is marked `@NonNullByDefault` so by definition the inner class inherits this annotation, and therefore `@NonNullByDefault` is not necessary explicitly on the inner class. However MVN raises an error that `@NonNullByDefault` must be applied. Whereas Eclipse raises an error that `@NonNullByDefault` must **_not_** be applied. And so I apply `@SuppressWarnings("null")` to make them both happy :)

```
@NonNullByDefault
public abstract class NeoHubAbstractDeviceData {

    @SuppressWarnings("null")
    @NonNullByDefault
    public abstract static class AbstractRecord {
```